darshan-util/pydarshan/tests/test_cffi_misc.py

```diff
@@ -62,3 +63,97 @@ def test_file_hash_type(log_path):
     # data type for the ids/hashes
     assert rec_counters["id"].dtype == np.uint64
     assert rec_fcounters["id"].dtype == np.uint64
+
+
+@pytest.mark.parametrize("dtype", ["numpy", "dict", "pandas"])
+def test_log_get_generic_record(dtype):
+    # regression test for issue #440
+    # see: https://github.com/darshan-hpc/darshan/issues/440
+
+    # collect the expected counter/fcounter column names
+    expected_counter_names = backend.counter_names("POSIX")
+    expected_fcounter_names = backend.fcounter_names("POSIX")
+
+    # assign the expected counter/fcounter values
+    expected_counter_vals = np.asarray(
```

Comment:
I can't think of a reason to use `asarray()` instead of plain `array()` here since you know you're providing a list an d not something that could be passed through unchanged; I'll leave that for you to cleanup in a follow-up